packages/http-server/src/server.ts

```diff
@@ -155,28 +155,38 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
     )();
   };
 
+  function setCommonCORSHeaders(incomingHeaders: IncomingHttpHeaders, res: ServerResponse) {
+    res.setHeader('Access-Control-Allow-Origin', incomingHeaders['origin'] || '*');
+    res.setHeader('Access-Control-Allow-Headers', incomingHeaders['access-control-request-headers'] || '*');
+    res.setHeader('Access-Control-Allow-Credentials', 'true');
+    res.setHeader('Access-Control-Expose-Headers', incomingHeaders['access-control-expose-headers'] || '*');
+  }
+
   const server = micri(
     Router.router(
       Router.on.options(
         () => opts.cors,
         (req: IncomingMessage, res: ServerResponse) => {
-          res.setHeader('Access-Control-Allow-Origin', req.headers['origin'] || '*');
-          res.setHeader('Access-Control-Allow-Headers', req.headers['access-control-request-headers'] || '*');
-          res.setHeader('Access-Control-Allow-Credentials', 'true');
-          res.setHeader('Access-Control-Allow-Methods', 'GET,DELETE,HEAD,PATCH,POST,PUT');
-          res.setHeader('Access-Control-Expose-Headers', req.headers['access-control-expose-headers'] || '*');
-          res.setHeader('Vary', 'origin');
-          res.setHeader('Content-Length', '0');
-          send(res, 204);
+          setCommonCORSHeaders(req.headers, res);
+
+          if (!!req.headers['origin'] && !!req.headers['access-control-request-method']) {
+            // This is a preflight request, so we'll respond with the appropriate CORS response
+            res.setHeader(
+              'Access-Control-Allow-Methods',
+              req.headers['access-control-request-method'] || 'GET,DELETE,HEAD,PATCH,POST,PUT,OPTIONS'
+            );
+
+            res.setHeader('Vary', 'origin');
```

Comment:
Good question is, whether we want this.
On one hand, `prism` sends the same reply regardless of `Origin`, so this is technically a(n innocent) lie.
On the other, I know most servers will actually include a `Vary` header in response to a preflight request, so I can totally understand why we'd want `prism` to do the same.

Reply:
This is coming from the previous middleware (fastify-cors) — I'm going to keep this for the time being